src/components/WearablePreview/EmoteControls.tsx

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable  @typescript-eslint/no-empty-function */
+import * as React from 'react'
+import { PreviewEmoteEventType } from '@dcl/schemas/dist/dapps/preview/preview-emote-event-type'
+import { IPreviewController } from '@dcl/schemas/dist/dapps/preview'
+import { Button } from '../Button/Button'
+import { Icon, WearablePreview } from '../..'
+import './EmoteControls.css'
+
+const ANIMATION_INTERVAL_PERCENTAGE = 0.1
+
+export type EmoteControlsProps = {
+  wearablePreviewId: string
+  className?: string
+  hideFrameInput?: boolean
+  hideProgressInput?: boolean
+  hidePlayButton?: boolean
+}
+
+type EmoteControlsState = {
+  frame: number
+  isPlaying: boolean
+  playingIntervalId?: number
+  previewController?: IPreviewController
+  length?: number
+}
+
+export class EmoteControls extends React.PureComponent<
+  EmoteControlsProps,
+  EmoteControlsState
+> {
+  state: EmoteControlsState = {
+    isPlaying: false,
+    frame: 0
+  }
+
+  async componentDidMount(): Promise<void> {
+    const previewController = WearablePreview.createController(
+      this.props.wearablePreviewId
+    )
+    const length = await previewController.emote.getLength()
+
+    previewController.emote.events.on(PreviewEmoteEventType.ANIMATION_END, () =>
+      this.setState({ isPlaying: false })
+    )
+
+    previewController.emote.events.on(
+      PreviewEmoteEventType.ANIMATION_PAUSE,
+      () => this.setState({ isPlaying: false })
+    )
+
+    previewController.emote.events.on(
+      PreviewEmoteEventType.ANIMATION_PLAY,
+      () => this.setState({ isPlaying: true })
+    )
+
+    this.setState({
+      previewController,
```

Comment:
even tho this works, I would advise against storing non-serializable data on the state (some dev tools might even throw warning on this). Could we move it from state to a class property? like `this.previewController = previewController` instead of using `setState`?

Reply:
good idea! moved to the class :)